pkg/apis/crd.projectcalico.org/v1/ippool.go

```diff
@@ -0,0 +1,140 @@
+type IPPoolSpec struct {
+	// The pool CIDR.
+	CIDR string `json:"cidr" validate:"net"`
+
+	// Contains configuration for VXLAN tunneling for this pool. If not specified,
+	// then this is defaulted to "Never" (i.e. VXLAN tunelling is disabled).
+	VXLANMode VXLANMode `json:"vxlanMode,omitempty" validate:"omitempty,vxlanMode"`
+
+	// Contains configuration for IPIP tunneling for this pool. If not specified,
+	// then this is defaulted to "Never" (i.e. IPIP tunelling is disabled).
+	IPIPMode IPIPMode `json:"ipipMode,omitempty" validate:"omitempty,ipIpMode"`
+
+	// When nat-outgoing is true, packets sent from Calico networked containers in
+	// this pool to destinations outside of this pool will be masqueraded.
+	NATOutgoing bool `json:"natOutgoing,omitempty"`
+
+	// When disabled is true, Calico IPAM will not assign addresses from this pool.
+	Disabled bool `json:"disabled,omitempty"`
+
+	// The block size to use for IP address assignments from this pool. Defaults to 26 for IPv4 and 112 for IPv6.
+	BlockSize int `json:"blockSize,omitempty"`
+
+	// Allows IPPool to allocate for a specific node by label selector.
+	NodeSelector string `json:"nodeSelector,omitempty" validate:"omitempty,selector"`
+
+	// Deprecated: this field is only used for APIv1 backwards compatibility.
+	// Setting this field is not allowed, this field is for internal use only.
+	IPIP *IPIPConfiguration `json:"ipip,omitempty" validate:"omitempty,mustBeNil"`
+
+	// Deprecated: this field is only used for APIv1 backwards compatibility.
+	// Setting this field is not allowed, this field is for internal use only.
+	NATOutgoingV1 bool `json:"nat-outgoing,omitempty" validate:"omitempty,mustBeFalse"`
+}
+
+type VXLANMode string
+
+const (
+	VXLANModeNever       VXLANMode = "Never"
+	VXLANModeAlways                = "Always"
+	VXLANModeCrossSubnet           = "CrossSubnet"
+)
+
+type IPIPMode string
+
+const (
+	IPIPModeNever       IPIPMode = "Never"
+	IPIPModeAlways               = "Always"
+	IPIPModeCrossSubnet          = "CrossSubnet"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IPPoolList contains a list of IPPool resources.
+type IPPoolList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []IPPool `json:"items"`
+}
+
+// NewIPPool creates a new (zeroed) IPPool struct with the TypeMetadata initialised to the current
+// version.
+func NewIPPool() *IPPool {
+	return &IPPool{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       KindIPPool,
+			APIVersion: "crd.projectcalico.org/v1",
+		},
+	}
+}
+
+// NewIPPoolList creates a new (zeroed) IPPoolList struct with the TypeMetadata initialised to the current
+// version.
+func NewIPPoolList() *IPPoolList {
+	return &IPPoolList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       KindIPPoolList,
+			APIVersion: "crd.projectcalico.org/v1",
+		},
+	}
+}
+
+type Mode string
+
+const (
+	Undefined   Mode = ""
```

Comment:
go formatting is off here?

Reply:
What do you think is off here?
It looks like it aligned on the `=`.
I tried removing the spaces in `Undefined   Mode` but my go fmt put them back.

Reply:
just looked weird to me, but I can believe that my eyes don't speak `gofmt`

Reply:
Are these constants actually used? The values actually look wrong to me.

Reply:
I think they're only used in the IPIPConfiguration, which itself is unused.